static/app/views/replays/utils/useReplayEvent.tsx

```diff
@@ -0,0 +1,180 @@
+import {useEffect, useState} from 'react';
+import type RRWebPlayer from 'rrweb-player';
+
+import {IssueAttachment} from 'sentry/types';
+import {Event} from 'sentry/types/event';
+import EventView from 'sentry/utils/discover/eventView';
+import {generateEventSlug} from 'sentry/utils/discover/urls';
+import RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+export type RRWebEvents = ConstructorParameters<typeof RRWebPlayer>[0]['props']['events'];
+
+type State = {
+  /**
+   * The root replay event
+   */
+  event: undefined | Event;
+
+  /**
+   * If any request returned an error then nothing is being returned
+   */
+  fetchError: undefined | RequestError;
+
+  /**
+   * If a fetch is underway for the requested root reply.
+   * This includes fetched all the sub-resources like attachments and `sentry-replay-event`
+   */
+  fetching: boolean;
+
+  /**
+   * The list of related `sentry-replay-event` objects that were captured during this `sentry-replay`
+   */
+  replayEvents: undefined | Event[];
+
+  /**
+   * The flattened list of rrweb events. These are stored as multiple attachments on the root replay object: the `event` prop.
+   */
+  rrwebEvents: undefined | RRWebEvents;
+};
+
+type Options = {
+  /**
+   * When provided, fetches specified replay event by slug
+   */
+  eventSlug: string;
+
+  /**
+   *
+   */
+  location: any;
+
+  /**
+   *
+   */
+  orgId: string;
+};
+
+type Result = {
+  // TODO: non-state?
+} & Pick<State, 'fetchError' | 'fetching' | 'event' | 'rrwebEvents' | 'replayEvents'>;
```

Comment:
I think we've preferred using interfaces over types for this reason https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

Reply:
ack